### PR TITLE
Add separate Server mapping in Gateway for each VirtualService

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Update Gatway server/VS mapping
  - Add a separate Server entry to the Gateway for each VS/IngressTrait so they
    are managed separately

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 1]
```diff
@@ -458,31 +524,33 @@ func (r *Reconciler) createOrUpdateGateway(ctx context.Context, trait *vzapi.Ing
...
-func (r *Reconciler) mutateGateway(gateway *istioclient.Gateway, trait *vzapi.IngressTrait, rule vzapi.IngressRule, secretName string) error {
+func (r *Reconciler) mutateGateway(gateway *istioclient.Gateway, trait *vzapi.IngressTrait, rule vzapi.IngressRule, vsName string, secretName string) error {
 	hosts, err := createHostsFromIngressTraitRule(r, rule, trait)
 	if err != nil {
 		return err
 	}
-	if len(gateway.Spec.Servers) > 0 {
-		hosts = appendToConfiguredHosts(hosts, gateway.Spec.Servers[0].Hosts)
+
+	r.Log.Infof("Mutating gateway %s for trait %s, vs: %s", gateway.Name, trait.Name, vsName)
```
Comment (kminder) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:533]: This seems more like a debug log.

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 2]
```diff
@@ -499,15 +567,31 @@ func (r *Reconciler) mutateGateway(gateway *istioclient.Gateway, trait *vzapi.In
...
-		_, hostFound := findHost(existingHosts, newHost)
-		if !hostFound {
-			existingHosts = append(existingHosts, strings.ToLower(newHost))
+// updateGatewayServersList Update/add the Server entry for the IngressTrait VirtualService to the gateway servers list
+//   - There will be a 1:1 mapping of Server-to-VirtualService
+func (r *Reconciler) updateGatewayServersList(existingServers []*istionet.Server, server *istionet.Server) []*istionet.Server {
+	servers := existingServers
+	if len(servers) == 0 {
+		servers = append(servers, server)
+		r.Log.Infof("Added new server for %s", server.Name)
```
Comment (kminder) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:576]: Is this too much logging?

File: application-operator/controllers/ingresstrait/ingresstrait_controller_test.go [hunk 3]
```diff
@@ -468,9 +468,10 @@ func TestSuccessfullyUpdateIngressWithCertSecret(t *testing.T) {
 		DoAndReturn(func(ctx context.Context, gateway *istioclient.Gateway, opts ...client.CreateOption) error {
 			assert.Equal(istionet.ServerTLSSettings_SIMPLE, gateway.Spec.Servers[0].Tls.Mode, "Wrong Tls Mode")
 			assert.Equal("cert-secret", gateway.Spec.Servers[0].Tls.CredentialName, "Wrong secret name")
+			assert.Len(gateway.Spec.Servers, 1)
 			assert.Contains(gateway.Spec.Servers[0].Hosts, "test-host", "doesn't contain expected host")
-			assert.Contains(gateway.Spec.Servers[0].Hosts, "test2-host", "doesn't contain expected host")
-			assert.Contains(gateway.Spec.Servers[0].Hosts, "test3-host", "doesn't contain expected host")
+			//assert.Contains(gateway.Spec.Servers[1].Hosts, "test2-host", "doesn't contain expected host")
```
Comment (kminder) [inline application-operator/controllers/ingresstrait/ingresstrait_controller_test.go:473]: Cleanup

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go (hunk 1 above)
Comment (michael-cico) [reply application-operator/controllers/ingresstrait/ingresstrait_controller.go:477]: yes, it is, I'll remove it on the next push.

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go (hunk 2 above)
Comment (michael-cico) [reply application-operator/controllers/ingresstrait/ingresstrait_controller.go:520]: I just don't want to have to flip the debug switch while I'm experimenting ;-)

File: application-operator/controllers/ingresstrait/ingresstrait_controller_test.go (hunk 3 above)
Comment (michael-cico) [reply application-operator/controllers/ingresstrait/ingresstrait_controller_test.go:473]: I have to re-work these tests once I settle on the solution.

Comment (michael-cico) [reply application-operator/controllers/ingresstrait/ingresstrait_controller_test.go:473]: And add some

**Pull request closed without merging**